Src/VimCore/CoreInterfaces.fs

```diff
@@ -2299,6 +2299,22 @@ type ModeArgument =
     /// match to process and the range is the full range to consider for a replace
     | Substitute of SnapshotSpan * SnapshotLineRange * SubstituteData
 
+with
+
+    // Running linked commands will throw away the ModeSwitch value.  This can contain
+    // an open IUndoTransaction.  This must be completed here or it will break undo in the
+    // ITextBuffer
+    member x.CompleteAnyTransaction =

```

Comment:
>CompleteAnyTransaction [](start = 13, length = 22)

Ha. I saw this and said "i know i've already defined this method" spent a few minutes looking for that code, found it then saw you'd already also found and refactored that code in this very PR ;)